controllers/vspherevm_controller.go

```diff
@@ -569,6 +570,9 @@ func (r vmReconciler) retrieveVcenterSession(ctx goctx.Context, vsphereVM *infra
 			params)
 	}
 
+	if cluster.Spec.InfrastructureRef == nil {
```

Comment:
This is the minimal fix for this, which I think should be backported. Otherwise there's a bit of refactoring that could be done on this controller to do this earlier in the reconcile, but there's a few questions around what should happen when the cluster is paused, and how the controller reacts if it can't find the cluster. 

Currently it ignores errors when trying to retrieve the cluster and only changes behaviour if it can find the cluster and the cluster is paused.

Happy to do that on this PR, but I thought it might be more difficult to refactor changes like that - @sbueringer @chrischdi opinions?

Reply:
I think this one is good for backporting. If we want to do the improved: let's do that on top and separate it from this so we don't cherry-pick it back.

Reply:
Sounds good. Let's keep this PR as is